{ 
  "text" : "A flaw was found in libgit2 before version 0.27.3. It has been discovered that an unexpected sign extension in git_delta_apply function in delta.c file may lead to an integer overflow which in turn leads to an out of bound read, allowing to read before the base object. An attacker may use this flaw to leak memory addresses or cause a Denial of Service.\n" ,
  "entities" : [ { 
    "eid" : "E1" ,
    "keyid" : "6255c3e0f88024e53b86048db1711cc1" ,
    "type" : "MiscEntity" ,
    "subType" : "function" ,
    "value" : "apply function" ,
    "begin" : 121 ,
    "end" : 151 ,
    "canonicalForm" : "apply function" ,
    "score" : "0.0"
  }
  , { 
    "eid" : "E2" ,
    "keyid" : "Vulnerability#80376fa50e037b74d2dee0090921aa65" ,
    "type" : "Vulnerability" ,
    "subType" : "Vulnerability" ,
    "value" : "integer overflow" ,
    "begin" : 167 ,
    "end" : 183 ,
    "canonicalForm" : "integer overflow" ,
    "score" : "0.0"
  }
  , { 
    "eid" : "E3" ,
    "keyid" : "TTP#9c98ef6a7451d6f82ca2194ee23d1acb" ,
    "type" : "TTP" ,
    "subType" : "TTP" ,
    "value" : "denial of service" ,
    "begin" : 336 ,
    "end" : 353 ,
    "canonicalForm" : "Denial of Service" ,
    "score" : "0.0"
  }
  ,  ] ,
  "relations" : [ { 
    "rid" : "R1" ,
    "type" : "lead" ,
    "predicate" : { 
      "begin" : 152 ,
      "end" : 160 ,
      "surfaceForm" : "may lead"
    }
     ,
    "arguments" : [ { 
      "argnum" : 1 ,
      "begin" : 121 ,
      "end" : 151 ,
      "eid" : "E1" ,
      "canonicalForm" : "apply function"
    }
    , { 
      "argnum" : 2 ,
      "begin" : 167 ,
      "end" : 183 ,
      "eid" : "E2" ,
      "canonicalForm" : "integer overflow"
    }
    ,  ]
  }
  ,  ]
}
